src/tests/example.spec.js

```diff
@@ -28,3 +27,154 @@ test.describe('Saucedemo app basic tests', () => {
     });
 });
+
+
+
+test.describe('Saucedemo UNIT 10', () => {
+        test.beforeEach( async (/** @type {{ app: import('../pages/Application').Application }} */{ app },) =>{
+            await app.login.navigate();
+            await app.login.performLogin('standard_user', 'secret_sauce');
+        } );
+
+
+        test('1. Check sorting by Name A Z', async ({app})=> {
+                  await app.inventory.sortField.selectOption('az');
+                  const goodsName = await app.inventory.inventoryItemsName.allTextContents();
+                  await expect([...goodsName].sort()).toEqual(goodsName);
+          });
+
+        test('1. Check sorting by Name Z A', async ({app})=> {
+                await app.inventory.sortField.selectOption('za');
+                const goodsName = await app.inventory.inventoryItemsName.allTextContents();
+ //             await expect([...goodsName].sort().reverse()).toEqual(goodsName);
+                await expect([...goodsName].sort((a, b) => b.localeCompare(a))).toEqual(goodsName);
+        });
+
+        test('1. Check sorting by Price Low to high', async ({app}) => {
+                await app.inventory.sortField.selectOption('lohi');
+                const goodsPriceText = await app.inventory.inventoryItemsPrice.allTextContents();
+                const goodsPrice = goodsPriceText.map(price => parseFloat(price.replace('$', '')));     
+                await expect([...goodsPrice].sort((a, b) => a - b)).toEqual(goodsPrice);
+        });
+
+        test('1. Check sorting by Price High to Low', async ({app}) => {
+                await app.inventory.sortField.selectOption('hilo');
+                const goodsPriceText = await app.inventory.inventoryItemsPrice.allTextContents();
+                const goodsPrice = goodsPriceText.map(price => parseFloat(price.replace('$', '')));     
+                await expect([...goodsPrice].sort((a, b) => b - a)).toEqual(goodsPrice);
+        });
+
+
+
+        test('test2 Add several random products to the Shopping Cart ana Verify  Name, Description, Price', async ({app}) =>{
+ //find quantity of goods on page
+                const goodsCount = await app.inventory.inventoryItems.count();
+//select 3 random goods and save their data
+                        function GoodsData (goodName, description, price){
+                                this.goodName = goodName;
+                                this.description = description;
+                                this.price= price;
+                                };
+                        let goodsArray=[];
+                        let selectedGoods = [];
+                        let randomGood = 0;
+                        for (let i=0; i<3; i++){
+                                        do {
+                                                randomGood = Math.floor(Math.random() * goodsCount) + 1;
+                                        } while (goodsArray.includes(randomGood));
```

Comment:
Do we really need do..while and for at the same time? Lets rewrite this code without while please.